extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -104,8 +62,8 @@ function SidebarWithRef(
   // We are using refs here to refer to common strings, objects, and functions used.
   // This is because they are needed within `useEffect` calls below (but are not depended for triggering)
   // We use `useValueRef` for props that might change (user-controlled)
+  const sideRef = useValueRef(sideProp);
```

Comment:
Hmm. I think here it actually should be a state? Does the component need to be rerendered when the side changes? It might need to be ignored for animations, but for rendering and component in general this could be a different story.

Reply:
You are correct.  The way we designed the `side` variable the order should be as follows when side is not defined -

1. 1st `useLayoutEffect` assigns `side` variable to `left` or `right` depending on document RTL status
2. 2nd `useLayoutEffect` (that handles animations) should exit early since `side` is not yet set
3. On re-render `side` is now set to `left` or `right`
4. 1st `useLayoutEffect` should exit early since `side` is already set
5. 2nd `useLayoutEffect` should be triggered by the `side` changing

So because of number 5, we do need to have `side` as a state variable.  Thanks!